### PR TITLE
Adding GetApproximateMemTableStats method

### PR DESCRIPTION
**Pull request by VitaliyLi**
The method returns not only size of memtables, but also approximation of number of entries. MyRocks needs this to produce more accurate query plans.

### CODE REVIEW

**Comment by siying** [comment]
A general question: do we need to return this stats as quick as possible, or as a background stats collecting job? If it is the later case, we can consider to iterate the memtable and get accurate count and size. Memtables are not that large, and one range can be even smaller.

**Comment by VitaliyLi** [comment]
Yeah, we need these stats quickly, as we use them during query evaluation.

**Comment by VitaliyLi** [comment]
Hm, apparently something got broken when I was cherry-picking the commit. Will fix the build issues.

**Comment by siying** [comment]
Can you add a unit test?

**Comment by siying** [comment]
Can you rebase and resolve the conflict?

File: db/db_impl.cc [hunk 1]
```diff
@@ -5544,6 +5544,29 @@ ColumnFamilyHandle* DBImpl::GetColumnFamilyHandle(uint32_t column_family_id) {
...
+  SuperVersion* sv = GetAndRefSuperVersion(cfd);
+
+  for (int i = 0; i < n; i++) {
+    // Convert user_key into a corresponding internal key.
+    InternalKey k1(range[i].start, kMaxSequenceNumber, kValueTypeForSeek);
+    InternalKey k2(range[i].limit, kMaxSequenceNumber, kValueTypeForSeek);
+    counts[i] = 0;
+    sizes[i] = 0;
+    auto memStats = sv->mem->ApproximateStats(k1.Encode(), k2.Encode());
+    auto immStats = sv->imm->ApproximateStats(k1.Encode(), k2.Encode());
```
Comment (siying) [inline db/db_impl.cc]: Here too. The auto is not recommended.

File: db/db_impl.cc [hunk 2]
```diff
@@ -5544,6 +5544,29 @@ ColumnFamilyHandle* DBImpl::GetColumnFamilyHandle(uint32_t column_family_id) {
   return cf_memtables->GetColumnFamilyHandle();
 }
 
+void DBImpl::GetApproximateMemTableStats(ColumnFamilyHandle* column_family,
+                                         const Range* range,
+                                         int n, uint64_t* counts,
+                                         uint64_t* sizes) {
+  auto cfh = reinterpret_cast<ColumnFamilyHandleImpl*>(column_family);
+  auto cfd = cfh->cfd();
```
Comment (siying) [inline db/db_impl.cc]: We follow Google C++ Style, in which this auto is not recommended as it's not clear about the type in this context: https://google.github.io/styleguide/cppguide.html#auto

File: db/db_impl.cc [hunk 3]
```diff
@@ -5544,6 +5544,29 @@ ColumnFamilyHandle* DBImpl::GetColumnFamilyHandle(uint32_t column_family_id) {
   return cf_memtables->GetColumnFamilyHandle();
 }
 
+void DBImpl::GetApproximateMemTableStats(ColumnFamilyHandle* column_family,
+                                         const Range* range,
+                                         int n, uint64_t* counts,
+                                         uint64_t* sizes) {
```
Comment (siying) [inline db/db_impl.cc]: Do MyRocks need to get the same of multiple ranges in one function call? If not, I suggest we keep it simple. Just support one range.

**Pull request closed without merging**